### PR TITLE
Add TypeScript type definitions

### LINKED ISSUE #59: Add TypeScript type definitions
We should be able to start from [something like this](https://github.com/namoscato/google-photos-vue/blob/138243a143e9a220526a24af1a1046e61d406ac1/src/types/gapi.d.ts).

### PR DESCRIPTION
**Pull request by namoscato**
Resolves #59

### CODE REVIEW

**Review by cedpoilly — APPROVED** [review]
Looks good to me. Let's push it & collect feedback 🙂

File: vue-gapi.d.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    logout: () => Promise<void>
+
+    /** Determines if the user is signed in via local storage. */
+    isAuthenticated: () => boolean
+
+    /** Determines if the user is signed in via Google. */
+    isSignedIn: () => Promise<boolean>
+
+    /** Accept the callback to be notified when the authentication status changes. */
+    listenUserSignIn: (callback: (isSignedIn: boolean) => any) => Promise<void>
```
Comment (fmeyertoens) [inline vue-gapi.d.ts:61]: Does it make sense for the callback to return a value (any)? I think it could be `void`.

**Review by fmeyertoens — CHANGES_REQUESTED** [review]
Thanks for making this PR.

File: vue-gapi.d.ts (hunk 1 above)
Comment (namoscato) [reply vue-gapi.d.ts:61]: Yeah, sounds good; I'll make that change.

**Review by fmeyertoens — APPROVED** [review]
LGTM

**Pull request merged**